packages/SwingSet/src/types-internal.js

```diff
@@ -1,13 +1,61 @@
 export {};
 
 /**
+ * The internal data that controls which worker we use (and how we use
+ * it) is stored in a WorkerOptions record, which comes in "local" and
+ * "xsnap" flavors.
+ *
+ * @typedef { { type: 'local' } } LocalWorkerOptions
+ * @typedef { { type: 'xsnap', bundleIDs: BundleID[] } } XSnapWorkerOptions
+ *  bundleIDs indicate the SES lockdown and supervisor/liveslots bundles to
+ *  evaluate into a new xsnap worker
+ * @typedef { LocalWorkerOptions | XSnapWorkerOptions } WorkerOptions
+ *
  * @typedef { string } VatID
  * @typedef { string } MeterID
  * @typedef { { meterID?: MeterID } } HasMeterID
- * @typedef { import('./types-external.js').DynamicVatOptionsWithoutMeter } DynamicVatOptionsWithoutMeter
+ * @typedef { import('./types-external.js').BaseVatOptions } BaseVatOptions
+ * @typedef { import('./types-external.js').HasManagerType } HasManagerType
+ * @typedef { { workerOptions: WorkerOptions } } HasWorkerOptions
+ * @typedef { import('./types-external.js').HasEnableDisavow } HasEnableDisavow
+ * @typedef { import('@agoric/swingset-liveslots').VatDeliveryObject } VatDeliveryObject
+ * @typedef { import('@agoric/swingset-liveslots').VatDeliveryResult } VatDeliveryResult
  *
  * // used by vatKeeper.setSourceAndOptions(source, RecordedVatOptions)
  *
- * @typedef { DynamicVatOptionsWithoutMeter & HasMeterID } InternalDynamicVatOptions
- * @typedef { import('./types-external.js').StaticVatOptions | ( InternalDynamicVatOptions & HasMeterID ) } RecordedVatOptions
+ * @typedef { BaseVatOptions & HasMeterID & HasManagerType } InternalDynamicVatOptions
+ * @typedef { BaseVatOptions & HasMeterID & HasWorkerOptions & HasEnableDisavow } RecordedVatOptions
+ *
+ * @typedef {{
+ *   enablePipelining?: boolean,
+ *   workerOptions: WorkerOptions,
+ *   metered?: boolean,
+ *   critical?: boolean,
+ *   enableDisavow?: boolean,
+ *   useTranscript?: boolean,
+ *   reapInterval?: number | 'never',
+ *   vatParameters: Record<string, unknown>,
+ *   virtualObjectCacheSize: number,
+ *   name: string,
+ *   compareSyscalls?: (originalSyscall: {}, newSyscall: {}) => Error | undefined,
+ *   sourcedConsole: Pick<Console, 'debug' | 'log' | 'info' | 'warn' | 'error'>,
+ *   meterID?: string,
+ *   enableSetup: boolean,
```

Comment:
Is ManagerOptions `enableSetup` really required, or should it be optional?
```suggestion
 *   enableSetup?: boolean,
```

Reply:
At this point it's required: `ManagerOptions` are only created by vat-loader.js `create()`, taking `RecordedVatOptions` as an input. That function always populates `enableSetup`, even if it was missing from the recorded options. That made the consumer of `ManagerOptions` simpler (or at least it stopped complaining about the types).

That probably suggests that many other `ManagerOptions` fields are required too, and could lose the `?`. `enablePipelining`, `enableDisavow`, `critical`, and others. Lemme see how many I can change.

Reply:
Turns out I can delete a bunch of stale ones that weren't causing problems because they were marked as optional. Once upon a time, RecordedVatOptions and ManagerOptions were the same thing, but now they're better separated.